tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -0,0 +1,508 @@
+              References      = [] }
+
+    let private fromFSharpErrorInfo (errors: FSharpErrorInfo[]) : (ErrorInfo list * ErrorInfo list) =
+        let toErrorInfo (e: FSharpErrorInfo) : ErrorInfo =
+            let errorNumber = e.ErrorNumber
+            let severity = e.Severity
+
+            let error = if severity = FSharpErrorSeverity.Warning then Warning errorNumber else Error errorNumber
+
+            { Error   = error
+              Range   =
+                  { StartLine   = e.StartLineAlternate
+                    StartColumn = e.StartColumn
+                    EndLine     = e.EndLineAlternate
+                    EndColumn   = e.EndColumn }
+              Message = e.Message }
+
+        errors
+        |> List.ofArray
+        |> List.distinctBy (fun e -> e.Severity, e.ErrorNumber, e.StartLineAlternate, e.StartColumn, e.EndLineAlternate, e.EndColumn, e.Message)
+        |> List.map toErrorInfo
+        |> List.partition  (fun e -> match e.Error with Error _ -> true | _ -> false)
+
+    let private adjustRange (range: Range) (adjust: int) : Range =
+        { range with
+                StartLine   = range.StartLine   - adjust
+                StartColumn = range.StartColumn + 1
+                EndLine     = range.EndLine     - adjust
+                EndColumn   = range.EndColumn   + 1 }
+
+    let Fsx (source: string) : CompilationUnit =
+        fsFromString source Fsx |> FS
+
+    let FSharp (source: string) : CompilationUnit =
+        fsFromString source Fs |> FS
+
+    let baseline (dir: string, file: string) : CompilationUnit =
+        match (dir, file) with
+        | dir, _ when String.IsNullOrWhiteSpace dir -> failwith "Baseline tests directory cannot be null or empty."
+        | _, file when String.IsNullOrWhiteSpace file -> failwith "Baseline source file name cannot be null or empty."
+        | _ ->
+            { Source         = Baseline (dir, file)
+              Options        = defaultOptions
+              OutputType     = Library
+              SourceKind     = Fs
+              Name           = None
+              IgnoreWarnings = false
+              References     = [] } |> FS
+
+    let CSharp (source: string) : CompilationUnit =
+        csFromString source |> CS
+
+    let withName (name: string) (cUnit: CompilationUnit) : CompilationUnit =
+        match cUnit with
+        | FS src -> FS { src with Name = Some name }
+        | CS src -> CS { src with Name = Some name }
+        | IL _ -> failwith "IL Compilation cannot be named."
+
+    let withReferences (references: CompilationUnit list) (cUnit: CompilationUnit) : CompilationUnit =
+        match cUnit with
+        | FS fs -> FS { fs with References = fs.References @ references }
+        | CS cs -> CS { cs with References = cs.References @ references }
+        | IL _ -> failwith "References are not supported in IL"
+
+    let withOptions (options: string list) (cUnit: CompilationUnit) : CompilationUnit =
```

Comment:
Any chance we can compose options: ?
FS { fs with Options = fs.Options @  options }

Reply:
It was intended to be composed, it's rather a bug, thanks! :)